### PR TITLE
Pull in upstream updates from cisagov/skeleton-ansible-role

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

In this pull request I merge in the develop branch from [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role).  (This repository is so old it predates the skeletons!)

I also remove some code that was a workaround for ansible/ansible#56583, since that issue has been resolved and the workaround is no longer needed.

Finally, I make the code behave correctly and pass tests fro Fedora 29 and 31.

## 💭 Motivation and Context

I wanted to remove the workaround, so I may as well bring this repo into the modern world as well.

## 🧪 Testing

All pre-commit hooks and GitHub checks pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by felddy — APPROVED** [review]
This is a time capsule!  ⏳ 💊

File: tasks/main.yml
Comment (dav3r) [inline tasks/main.yml:54]: Is there a reason this `regex` and the one above it were not switched from single quotes to double quotes or was this just an oversight?

**Review by dav3r — APPROVED** [review]
👍 though see my one question.

File: tasks/main.yml
Comment (jsf9k) [reply tasks/main.yml:54]: What is there is correct.  This is YAML foo.

See [here](https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html), @dav3r.  I needed the double quotes for the strings containing escapes (`\t`).  The regexes don't need double quotes since they don't contain escapes; `\s` is a regex ~thing~thang, not an escape.

Let me know if this satisfies your question.  I will wait to merge until I hear back from you.

Comment (dav3r) [reply tasks/main.yml:54]: Yes, that makes sense, but it looks odd because we have a seemingly-inconsistent mix of single-quoting and double-quoting throughout this file (more than just these two regex lines).

Comment (jsf9k) [reply tasks/main.yml:54]: Well it looks odd, but thus is YAML.  I hesitate to add comments to these lines since I think a decent understanding of YAML's warts should be assumed when working with Ansible.

I will add a comment if you can convince @felddy or @mcdonnnj, though.  I will defer to the wisdom of the herd.

Comment (dav3r) [reply tasks/main.yml:54]: I agree with you and don't think that a comment is needed here either.  Although if we are only double-quoting when necessary, then shouldn't these lines be using single quotes?
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L53
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L63
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L65
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L67

Comment (mcdonnnj) [reply tasks/main.yml:54]: I agree with @dav3r about consistency. I think double quotes and escaping the backslash (`"\\s*rotate \\d+$"`) is more productive than a comment explaining single vs double quotes if we went that route however.

Comment (jsf9k) [reply tasks/main.yml:54]: @dav3r and @mcdonnnj, try on commit 30725fe for size.  Let me know what you think.

Comment (dav3r) [reply tasks/main.yml:54]: That works for me- thanks and sorry for the extra work!

Comment (jsf9k) [reply tasks/main.yml:54]: Once @mcdonnnj is happy I will merge.

Comment (jsf9k) [reply tasks/main.yml:54]: @mcdonnnj confirmed to me that he is happy with the quotes.

**Pull request merged**
